src/main/java/alfio/manager/user/UserManager.java

```diff
@@ -346,4 +394,50 @@ public Integer createPublicUserIfNotExists(String username, String email, String
         return userRepository.findIdByUserName(username).orElse(null);
     }
 
+
+    private void checkIsAdmin(Principal principal) {
+        if (principal == null) {
+            return;
+        }
+        if (isAdmin(findUserByUsername(principal.getName()))) {
+            return;
+        }
+        log.warn("User {} is not an admin", principal.getName());
+        throw new IllegalArgumentException("User " + principal.getName() + " is not an admin");
+    }
+
+    private void checkAccessToUserId(Principal principal, int userId) {
+        if (principal == null) {
+            return;
+        }
+        var currentUser = findUserByUsername(principal.getName());
+        if (isAdmin(currentUser)) {
+            return;
+        }
+        var targetUser = findUser(userId);
+        var targetUserOrgs = findUserOrganizations(targetUser.getUsername());
+        for (var org : targetUserOrgs) {
```

Comment:
todo: ad assert that the user is in only one user organization (which is actually always the case now).